tests/ui/manual_ignore_case_cmp.rs

```diff
@@ -0,0 +1,71 @@
+#![allow(clippy::all)]
+#![deny(clippy::manual_ignore_case_cmp)]
+
+use std::ffi::{OsStr, OsString};
+
+fn main() {}
+
+fn variants(a: &str, b: &str) -> bool {
+    if a.to_ascii_lowercase() == b.to_ascii_lowercase() {
+        return true;
+    }
+    if a.to_ascii_uppercase() == b.to_ascii_uppercase() {
+        return true;
+    }
+    let r = a.to_ascii_lowercase() == b.to_ascii_lowercase();
+    let r = r || a.to_ascii_uppercase() == b.to_ascii_uppercase();
+    r && a.to_ascii_lowercase() == b.to_uppercase().to_ascii_lowercase()
+}
+
+fn unsupported(a: char, b: char) {
+    // TODO:: these are rare, and might not be worth supporting
+    a.to_ascii_lowercase() == char::to_ascii_lowercase(&b);
+    char::to_ascii_lowercase(&a) == b.to_ascii_lowercase();
+    char::to_ascii_lowercase(&a) == char::to_ascii_lowercase(&b);
```

Comment:
Not really worth supporting. I would expect them to still trigger the lint though :thinking:

Reply:
oki, will keep them as part of tests to track progress if they ever change, but won't fix them in the lint